replica/database.cc

```diff
@@ -2347,18 +2347,23 @@ future<> database::flush_keyspace_on_all_shards(sharded<database>& sharded_db, s
     });
 }
 
-future<> database::snapshot_table_on_all_shards(sharded<database>& sharded_db, std::string_view ks_name, sstring table_name, sstring tag, bool skip_flush) {
+future<> database::snapshot_table_on_all_shards(sharded<database>& sharded_db, std::string_view ks_name, sstring table_name, sstring tag, db::snapshot_ctl::snap_views snap_views, bool skip_flush) {
     if (!skip_flush) {
         co_await flush_table_on_all_shards(sharded_db, ks_name, table_name);
     }
     auto uuid = sharded_db.local().find_uuid(ks_name, table_name);
     auto table_shards = co_await get_table_on_all_shards(sharded_db, uuid);
     co_await table::snapshot_on_all_shards(sharded_db, table_shards, tag);
+    if (snap_views) {
```

Comment:
Should this be ignored when this table is a view itself?

Reply:
In practice, it shouldn't be set, because when the upper level walks over all tables via the keyspace metadata (base tables and views), it passed snaps_views::no, and in the recursion path, when we take a snapshot of the views we pass snaps_views::no too. And even if it was set to snap_views::yes, `table_shards[this_shard_id()]->views()` should be empty, so the current implementation is harmless.

Reply:
Yes, I thought so too (that it is harmless no-op at worst) but wanted to confirm.